supervisor/hyperpod.go

```diff
@@ -90,13 +237,21 @@ func createHyperPod(f factory.Factory, spec *specs.Spec) (*HyperPod, error) {
 	}
 	glog.V(1).Infof("result: code %d %s\n", Response.Code, Response.Cause)
 
-	return &HyperPod{
+	hp := &HyperPod{
 		userPod:    userPod,
 		podStatus:  podStatus,
 		vm:         vm,
 		Containers: make(map[string]*Container),
 		Processes:  make(map[string]*Process),
-	}, nil
+	}
+
+	// create Listener process running in its own netns
+	if err = hp.startNsListener(); err != nil {
+		glog.V(1).Infof("%s\n", err.Error())
```

Comment:
when vm.StartPod() fails. the original code doesn't kill the vm. it is a bug blamed to @laijs.
you can also add it.